### PR TITLE
Update `Header` with new design, add additional basic functionality.

### PR DESCRIPTION
**Pull request by HaresMahmood**
# Overview

Update `Header` with new design, add additional basic functionality.

## Features

- Updated design.
   - Including `SearchBar`-component.
   > Note:
      Currently non-functional until search API is added and implemented, missing icon until the team as a whole decides on which icon pack to use.
- Updated logo.
- Basic navigation.
- Responsive design.
   > Note:
      Overflow menu for mobile yet to be created and populated.

# Screenshots

|                                                      Desktop                                                     |                                                      Mobile                                                      |
|:---------------------------------------------------------------------------------------------------------------:|:---------------------------------------------------------------------------------------------------------------:|
| image | image |

### CODE REVIEW

File: src/components/common/layout/header/index.tsx [hunk 1]
```diff
@@ -1,14 +1,40 @@
...
-        />
+    <header className="flex flex-row justify-between fixed top-0 z-50 w-full h-20 items-center bg-black/25 px-6 backdrop-blur-xl backdrop-filter">
+        <div className="flex gap-10">
+            <Image
+                src="/logo.svg"
+                alt="logo"
+                width={175}
+                height={27}
+                // priority
+            />
```
Comment (hreyesm) [inline src/components/common/layout/header/index.tsx:16]: Is this where the custom `Image` didn't work as expected?

**Review by hreyesm — APPROVED** [review]

File: src/components/common/layout/header/index.tsx (hunk 1 above)
Comment (HaresMahmood) [reply src/components/common/layout/header/index.tsx:16]: Essentially yeah, I think the `width` and `height` `props` don't affeect `.svg`s for some reason. I also didn't know how to implement the `...rest` `prop` properly.

Comment (hreyesm) [reply src/components/common/layout/header/index.tsx:16]: I'll take a look at this.

**Comment by ZeyadTarekk** [comment]
Try to resolve the conflicts @HaresMahmood to merge it

**Pull request merged**